hw04-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,49 @@
+package homework;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private final Map<Customer, String> map = new TreeMap<>((o1, o2) ->
+        (int) (o1.getScores() - o2.getScores())
+    );
+
+    public Map.Entry<Customer, String> getSmallest() {
+        for (Map.Entry<Customer, String> entry : map.entrySet()) {
+            return entryCopy(entry);
+        }
+        return null;
+    }
+
+    private Map.Entry<Customer, String> entryCopy(Map.Entry<Customer, String> entry) {
+        return Map.entry(
+            new Customer(entry.getKey().getId(), entry.getKey().getName(), entry.getKey().getScores()),
+            entry.getValue()
```

Comment:
Совсем мелочь, но тем не менее:

```suggestion
        var customer = entry.getKey();
        return Map.entry(
                new Customer(customer.getId(), customer.getName(), customer.getScores()),
                entry.getValue()
        );
```

Reply:
Исправил